pylint/checkers/base.py

```diff
@@ -2399,6 +2405,27 @@ def _is_singleton_const(node) -> bool:
             args=("'{}'".format(root_node.as_string()), suggestion),
         )
 
+    def _check_NaN_comparison(
+        self, left_value, right_value, root_node, checking_for_absence: bool = False
+    ):
+        nan_values = ("float('nan')", "numpy.NaN", "np.NaN")
+
+        def _is_NaN(node) -> bool:
+            return node.as_string() in nan_values
```

Comment:
This way of checking the Nan value is not robust. What if, for example, `numpy` is imported as `nmp`?
Instead we should ask astroid to infer the value of the node.
For example in case of `float('nan')` we can use astroid this way:
```
 if isinstance(node, astroid.Call) and node.args[0].value == 'nan' and node.inferred()[0].pytype() == 'builtins.float':
     return True
```

Reply:
Okay. will be fixed.